karapace/schema_registry_apis.py

```diff
@@ -446,15 +446,16 @@ async def schemas_list(self, content_type: str, *, request: HTTPRequest, user: U
             if self._auth and not self._auth.check_authorization(user, Operation.Read, f"Subject:{subject}"):
                 continue
             for schema_version in schema_versions:
-                response_schemas.append(
-                    {
-                        "subject": schema_version.subject,
-                        "version": schema_version.version,
-                        "id": schema_version.schema_id,
-                        "schemaType": schema_version.schema.schema_type,
-                        "schema": schema_version.schema.schema_str,
-                    }
-                )
+                response_schema = {
+                    "subject": schema_version.subject,
+                    "version": schema_version.version,
+                    "id": schema_version.schema_id,
+                    "schemaType": schema_version.schema.schema_type,
+                }
+                if schema_version.references:
+                    response_schema["references"] = [r.to_dict() for r in schema_version.references]
+                response_schema["schema"] = schema_version.schema.schema_str
```

Comment:
question: Why not set this in the dict literal above?

Reply:
I tried to keep ordering of keys. Actually elsewhere our json responses are sorted by key, not sure if it was intentional or not.